### ISSUE TITLE
offload-build: issues with VCS sources not being updated

### ISSUE THREAD

**Issue by dvzrv** [archlinux/devtools#77]
When using VCS sources (e.g. a signed tag, a pinned commit), offload-build does not update the local bare repo, which may lead to e.g. erroring on:

```
git repo ... SIGNATURE NOT FOUND
```

After updating the local bare repository of the upstream sources, the offload-build can commence as normal.

**Comment by anthraxx** [comment]
This is actually a bug in `makepkg` when calling `makepkg --source` without having fetched the git sources of the new tag beforehand. We can hack and work around it in an ugly way, but this should rather be reporter where the bug actually is.

**Comment by Foxboron** [comment]
It seems like `download_git` is never run unless you pass `makepkg --allsource`. Is there any cons using `--allsource` instead of `--source`?

**Comment by anthraxx** [comment]
@Foxboron yes: we do not want to bundle all remote sources into the source ball to repeatedly upload them on each build to the buildserver.

**Comment by Foxboron** [comment]
This needs some more work in `makepkg`. There is no way to say "only download VCS sources" in the current implementation.

**Comment by anthraxx** [comment]
Or frankly less work ;-) This is IMO lot of complexity for creating a 
source tarball that is actually unexpected from a user perspective. We 
should just not validate *remote* content at all and just tar together 
the sources as advertised.
